src/app/translate.tsx

```diff
@@ -0,0 +1,220 @@
+"use client";
+
+import React, { useEffect, useState, useCallback } from "react";
+import defaultTranslations from "@/i18n/translations.en.json";
+import classes from "@/components/header.module.css";
+
+const languages = {
+  en: "English (United States)",
+  ar: "العربية",
+  ca: "Català",
+  "zh-CN": "简体中文",
+  "zh-TW": "繁體中文",
+  cs: "Čeština",
+  nl: "Nederlands",
+  fr: "Français (France)",
+  de: "Deutsch",
+  he: "עברית (ישראל)",
+  hu: "Magyar",
+  id: "Indonesian",
+  it: "Italiano (Italia)",
+  ja: "日本語",
+  ko: "한국어",
+  pl: "Polski (Polska)",
+  "pt-PT": "Português (Portugal)",
+  "pt-BR": "Português brasileiro",
+  ro: "Romanian",
+  ru: "Русский",
+  sk: "Slovenčina (Slovensko)",
+  "es-ES": "Español",
+  "sv-SE": "Svenska",
+  tr: "Türkçe",
+  uk: "Українська",
```

Comment:
This hard-coded list of languages and language codes should exactly match the extensions for the JSON language files we add.

Due to the way the logic is, it's probably best we use shorter codes whenever possible, so we may want to change:

"es-ES" to "es"
"sv-SE" to "sv"
"pt-PT" to "pt"
"zh-CN" to "zh"